app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java

```diff
@@ -221,10 +225,48 @@ public static void newVideoLoaded(@NonNull String videoId) {
      */
     public static void onComponentCreated(@NonNull Object conversionContext, @NonNull AtomicReference<Object> textRef) {
         try {
-            if (!SettingsEnum.RYD_ENABLED.getBoolean()) return;
+            Object original = textRef.get();
+            if (original instanceof Spanned) {
+                SpannableString replacement = getDislikeSpanForContext(conversionContext, (Spanned) original);
+                if (replacement != null) {
+                    textRef.set(replacement);
+                }
+            }
```

Comment:
It would probably be better to introduce a new "injection point" so that you can avoid the instance check if I understood the current implementation correctly.

Reply:
I still don't understand why the check is necessary. The places this method is called from should all be returning a span.

Reply:
The path builder string can be checked before casting. If it matches the only type can be the expected.

Reply:
Can the order of operation be switched here?

Reply:
I didn't understand the problem yet with rearrangement. Why would it be simpler to keep the context check after the atomic reference casting

Reply:
Alright, what was the problem with casting `textRef` here:

`onComponentCreated(Object conversionContext, AtomicReference<CharSequence> textRef)`

Can the atomic reference hold anything than a nonnull `CharSequence`?

Reply:
In that case you can move the hook down until there is a null check present in the code where you inject the method. That way you can be sure it is a non null char reference.

Reply:
Your second injection point hooks the char sequence and returns the modified one. Is the same not possible for the first hook? Basically do we really depend on the atomic reference?

Reply:
I think it might be possible to copy the conversion context register to another register that is not yet used/necessary; that way, you would have access to it when the initial register it occupied is used.

Reply:
In that case, would a null check suffice instead of an instance check in the integrations?